app/controllers/comments/list.js

```diff
@@ -0,0 +1,77 @@
+import { db } from "~/utils/db.server";
+import {
+    INVALID_PARAMS_FOR_OPERATION_ERROR_MESSAGE,
+  } from "~/utils/constants";
+import { DEFAULT_LIMIT, DEFAULT_OFFSET } from "~/utils/backend/constants";
+import { canEditComment } from "~/utils/backend/comments";
+
+const findValue = (element,userId,value) => {
+    return element.user == userId && element.value == value;
+}
+
+const reorderCommentsByVotes = (first, second) => {
+    if((first.approvedBy !== null && second.approvedBy === null)||
+       (first.votes > second.votes)||
+       (first.recent_activity > second.recent_activity)){
+        return  -1;
+    }
+    if((first.approvedBy === null && second.approvedBy !== null)|| 
+       (first.votes < second.votes)||
+       (first.recent_activity < second.recent_activity)){
+        return  1;
+    }
+    return 0;
+}
+
+const reorderCommentsByActivity = (first, second) => {
+    if((first.approvedBy !== null && second.approvedBy === null)||
+       (first.recent_activity > second.recent_activity)){
+        return  -1;
+    }
+    if((first.approvedBy === null && second.approvedBy !== null)|| 
+       (first.recent_activity < second.recent_activity)){
+        return  1;
+    }
+    return 0;
+}
+
+export const listComments = async (params) => {
+const {questionId, limit, offset, userEmail , userId, sortBy, sessionToken} = params
+
+
+if(!questionId || questionId < 1 ){
+    return {
+        error: {
+            message: INVALID_PARAMS_FOR_OPERATION_ERROR_MESSAGE,
+            details: 'The question id must be an integer not minor to 1',
+        }
+    }
+}
+
+const fetchComments = await db.Comments.findMany({
+    where: {
+        questionId: questionId
+    },
+    take: limit || DEFAULT_LIMIT,
+    skip: offset || DEFAULT_OFFSET,
+    include:{
+        approver: true,
+        user: true,
+        CommentVote:true,
+    }
+});
+
+const comments =  fetchComments.map((comment) => {
+    comment.canEdit = canEditComment(comment, userEmail, sessionToken);
+    delete comment.sessionHash;
+    comment.votes = comment.CommentVote.reduce((acc, current)=> {
+        return acc + current.value
+    }, 0);
+    comment.has_upvote = comment.CommentVote.some((comment) => findValue(comment, userId, 1));
+    comment.has_downvoted =  comment.CommentVote.some((comment) => findValue(comment, userId, -1));
+    comment.recent_activity = comment.updatedAt > comment.createdAt ? comment.updatedAt : comment.createdAt;
+    return comment;
+});
+sortBy === "votes" ? comments.sort(reorderCommentsByVotes) : comments.sort(reorderCommentsByActivity);;
```

Comment:
Amazing work Gina, congrats! 🎊 
I just had a comment regarding this line, the sorting of comments, here it's done with the sort JS method, however in the old Express project, for the comment list controller we set the sorting at the ORM level ([reference for the code I'm talking about](https://github.com/wizeline/wize-q-back/blob/81c9981222793119192d73cccf3f9a164db66bda/controllers/comments.js#L60-L71))
Since MySQL is well optimized to filter and sort the records I think it would be more efficient to keep it that way at the ORM level just as an advice, if its troublesome or complex to do it that way with Prisma I don't think it matters that much given the volume of data we handle in the application so it's just a thought, I'm approving the PR, thank you!

Reply:
I can do the orderBy at the ORM level for votes and approveBy, but for recent activity I haven't found a way to do it since it's a custom field.
What can be done is that at the moment of creating the comment we can also fill the updateat field and so I can  use that field to sort by recent activity.

Reply:
No worries, I imagined it was because of the recentActivity, I haven't found a way to attach a property to a model using a query literal either like we used to do with Sequelize, we could even leave this for a future item as an enhancement, so no rush, you can complete the PR if you want, amazing job again 🥇

Reply:
Found this Prisma feature that we can use to automatically update updatedAt columns https://www.prisma.io/docs/reference/api-reference/prisma-schema-reference#updatedat . It could be useful as we dont have to manually update it. Once implemented, we can use this value for sorting as mentioned 😄

Reply:
New ticket for it #45